sql/src/main/java/io/crate/analyze/InsertFromValuesAnalyzer.java

```diff
@@ -607,4 +609,19 @@ private void addGeneratedPartitionedColumnValue(ColumnIdent columnIdent,
                 givenValue, generatedValue));
         }
     }
+
+    static void verifyOnConflictTargets(List<String> constraintColumns, List<ColumnIdent> pkColumnIdents) {
+        if (constraintColumns.size() != pkColumnIdents.size()) {
+            throw new IllegalArgumentException(
+                String.format("Number of conflict targets (%s) did not match the number of primary key columns (%s)",
+                    constraintColumns, pkColumnIdents));
+        }
+        for (ColumnIdent pkColIdent : pkColumnIdents) {
+            if (!constraintColumns.contains(pkColIdent.name())) {
```

Comment:
what if the primary key is on a column inside an object? `create table t (obj object as (x int primary key))`

Reply:
I didn't know we supported that. We can also check the path.